tools/make_patch.c

```diff
@@ -108,9 +108,21 @@ int create_symbol(struct symbol **tip, const int value, const char *name)
 const struct symbol *find_symbol(const struct symbol *symbols, const char *name)
 {
 	const struct symbol *symbol = symbols;
+	int namelen = strlen(name);
+	int symbolnamelen = 0;
+	const char *equalLastCharacters = NULL;
 	while (symbol) {
-		if (strcmp(symbol->name, name) == 0) {
-			return symbol;
+		symbolnamelen = strlen(symbol->name);
+		if (namelen > symbolnamelen) {symbol = symbol->next; continue;}
+		equalLastCharacters = &symbol->name[symbolnamelen - namelen];
+		if (name[0] == '.') { // We are looking for a local label.
+			if (strcmp(equalLastCharacters, name) == 0) {
+				return symbol;
+			} 
+		} else {
+			if (strcmp(symbol->name, name) == 0) {
+					return symbol;
+			}
```

Comment:
```c
// If we're looking for a local label, only compare the part starting from the period
char *compare = symbol->name;
if (name[0] == '.') compare += symbolnamelen - namelen;
if (strcmp(compare, name) == 0) return symbol;
```

Reply:
Implemented you code changes in https://github.com/pret/pokecrystal/pull/882/commits/74142c471fc03d09c01ea35bfd59952b69b612c1